tests/test-output-flex-way-relation-add.cpp

```diff
@@ -0,0 +1,387 @@
+#include <catch.hpp>
+
+#include "common-import.hpp"
+#include "common-options.hpp"
+
+static testing::db::import_t db;
+
+static char const *const conf_file = "test_output_flex_way.lua";
+
+static const char *const tdata[] = {
+    "n10 v1 dV x10.0 y10.0",
+    "n11 v1 dV x10.0 y10.1",
+    "n12 v1 dV x10.1 y10.0",
+    "n13 v1 dV x10.1 y10.1",
+    "n14 v1 dV x10.2 y10.0",
+    "n15 v1 dV x10.2 y10.1",
+    "n16 v1 dV x10.3 y10.0",
+    "n17 v1 dV x10.3 y10.1",
+    "n18 v1 dV x10.4 y10.0",
+    "n19 v1 dV x10.4 y10.1",
+    "w11 v1 dV Tt1=yes Nn12,n13",
+    "w12 v1 dV Tt2=yes Nn14,n15",
+    "w13 v1 dV Ttboth=yes Nn16,n17",
+    "w14 v1 dV Ttboth=yes Nn18,n19",
+    "w15 v1 dV Tt=ag Nn17,n19",
+    "r30 v1 dV Tt=ag Mw10@,w11@,w12@mark,w13@,w14@mark"};
+
+// adding relation (marked), also check case where relation changed to contain
+// the way under test
+
+TEST_CASE("test way: add relation with way in t1 (marked)")
+{
+    options_t options = testing::opt_t().slim().flex(conf_file);
+
+    testing::data_t data{tdata};
+
+    data.add("w10 v1 dV Tt1=yes,t2=yes Nn10,n11");
+
+    SECTION("no relation") {}
+    SECTION("relation without w10") { data.add("r32 v1 dV Tt=ag Mw15@"); }
```

Comment:
We are missing the indirect dependencies through members here. r32 should have the other ways as members, when it gets changed. This should then cover the last error case we have seen (the one that resulted in #1244). Unfortunately that means that we need separate functions. :(